### ISSUE TITLE
Can't use an undefined value as a HASH reference at /usr/local/cpanel/scripts/elevate-cpanel line 2470.

### ISSUE THREAD

**Issue by ggrigon** [cpanel/elevate#144]
Hello,

I just used this script to elevate a **DNS Only** install from **CentOS 7.9** to **AlmaLinux 8.6**. cPanel version **106.9**.

Even with this error message on the script:

```
The elevation process failed during stage 4.

You can continue the process after fixing the errors by running:

    /usr/local/cpanel/scripts/elevate-cpanel --continue

You can check the error log by running:

    /usr/local/cpanel/scripts/elevate-cpanel

Last Error:

Can't use an undefined value as a HASH reference at /usr/local/cpanel/scripts/elevate-cpanel line 2470.
```

It seems to be running fine, but I saw the bind didn't start correctly, giving startup errors.

Got it running fine after did a 
```
/scripts/rebuilddnsconfig
```

**Comment by jimsefton** [comment]
Interesting, I had about 6 or 7 of those errors. Couldn't find anything about it so I forced the script to continue by adding a return before the error line. Seems to all be working but bit worried about doing my other boxes. The other lines were re-install of Immunify and checking Litespeed (and some others). Did you ever get to the bottom of it?

**Comment by ggrigon** [comment]
Hello [jimsefton](https://github.com/jimsefton),

This server was a DNSOnly version, the only issue I've encountered was that.
It's running fine since that.

**Comment by toddr** [comment]
looks like we're getting an empty hash from reinstall. Need to fix this assumption.

```perl
my $product_type = read_stage_file('reinstall')->{'imunify'} || return;
```

**Comment by jimsefton** [comment]
When I ran the script I had this immunify error, but I also had 6 or 7 others that had similar read_stage_file references. Not sure if you can check those too?

**Comment by atoomic** [comment]
By chance, do you have access to the output log so I can see these errors?
thanks

**Comment by atoomic** [comment]
I can see multiple locations in the codebase where similar issues could occurred.
I provided a more generic fix via PR #153 which should avoid such issues in the future
